### PR TITLE
[FEAT][TNA] Fabric Component - Add Component to App (Android)

### PR DESCRIPTION
**Pull request by cipolleschi**
This PR adds the instruction to connect a Fabric Component to the Android App.

### CODE REVIEW

File: docs/the-new-architecture/pillars-fabric-components.md [hunk 1]
```diff
@@ -531,6 +531,11 @@ cd android
 
 This script first adds the package to the app, in the same way iOS does. Then, after moving to the `android` folder, it invokes a `gradlew` task to generate the codegen.
 
+:::note
+To run the codegen, you need to enable the **New Architecture** in the Android app. This can be done by opening the `gradle.properties` files and by switching the `newArchEnabled` property from `false` to `true`.
```
Comment (cortinico) [inline docs/the-new-architecture/pillars-fabric-components.md:535]:
```suggestion
To run the CodeGen, you need to enable the **New Architecture** in the Android app. This can be done by opening the `gradle.properties` files and by switching the `newArchEnabled` property from `false` to `true`.
```

File: docs/the-new-architecture/pillars-fabric-components.md [hunk 2]
```diff
@@ -770,7 +775,64 @@ This command will look for all the dependencies of the project and it will insta
...
+   ```diff
+   include $(REACT_ANDROID_DIR)/Android-prebuilt.mk
+
+   # If you wish to add a custom TurboModule or Fabric component in your app you
+   # will have to include the following autogenerated makefile.
+   # include $(GENERATED_SRC_DIR)/codegen/jni/Android.mk
+
+   +include $(NODE_MODULES_DIR)/rnt-centered-text/android/build/generated/source/codegen/jni/Android.mk
+   include $(CLEAR_VARS)
+   ```
```
Comment (cortinico) [inline docs/the-new-architecture/pillars-fabric-components.md:800]: Commenting here but valid for the rest of the PR. You seem to have 3 space indentation here which I would remove

File: docs/the-new-architecture/pillars-fabric-components.md [hunk 3]
```diff
@@ -770,7 +775,64 @@ This command will look for all the dependencies of the project and it will insta
...
+
+1. In the same file, scroll down until you find a list of `libreact` libraries. There, we have to add the the library that has been generated. To do so, we need to add this line:
+   ```diff
+   libreact_codegen_rncore \
+   +react_codegen_RTNCenteredText \
```
Comment (cortinico) [inline docs/the-new-architecture/pillars-fabric-components.md:805]:
```suggestion
   +libreact_codegen_RTNCenteredText \
```

File: docs/the-new-architecture/pillars-fabric-components.md [hunk 4]
```diff
@@ -770,7 +775,64 @@ This command will look for all the dependencies of the project and it will insta
...
+   libreact_debug \
+   ```
+
+:::note
+The exact name of the library can be found in the generated `Android.mk` that is located in the folder:
+`node_modules/rnt-centered-text/android/build/generated/source/codegen/jni/Android.mk`.
+The name of the library is the value of the `LOCAL_MODULE` variable of that file.
```
Comment (cortinico) [inline docs/the-new-architecture/pillars-fabric-components.md:812]: I would not ask users to look into the generated Makefile. I think we should rephrase this to either:
- Mention that the library will be `libreact_codegen_<libraryname>` where `libraryname` is the one specified in the config
- Mention that this is not needed as libs will be autolinked

**Review by cortinico — APPROVED** [review]

File: docs/the-new-architecture/pillars-fabric-components.md (hunk 2 above)
Comment (cipolleschi) [reply docs/the-new-architecture/pillars-fabric-components.md:800]: The linter forced me to add those 3 spaces. 
Some indentation is needed not to break the numbered list, and then the linter forces me to use a specific number of spaces.

**Review by cortinico — APPROVED** [review]
